clang-tools-extra/clang-doc/tool/ClangDocMain.cpp

```diff
@@ -127,16 +135,85 @@ std::string getFormatString() {
 // GetMainExecutable (since some platforms don't support taking the
 // address of main, and some platforms can't implement GetMainExecutable
 // without being given the address of a function in the main executable).
-std::string GetExecutablePath(const char *Argv0, void *MainAddr) {
+std::string getExecutablePath(const char *Argv0, void *MainAddr) {
   return llvm::sys::fs::getMainExecutable(Argv0, MainAddr);
 }
 
+llvm::Error getAssetFiles(clang::doc::ClangDocContext &CDCtx) {
+  std::error_code Code;
+  llvm::SmallString<128> FilePath(UserAssetPath);
+  for (DirIterator DirIt = DirIterator(UserAssetPath, Code),
+                   DirEnd = DirIterator();
```

Comment:
```suggestion
                   DirEnd;
```
This doesn't need to be initialized that way, and appears to be the common pattern in clang and elsewhere when using directory_iterator. I'm fine w/ using the iterator from VFS, but usage should probably be consistent w/ what's used in other clang-tools-extra projects. If nothing else, there's an argument for being consistent w/ clang.